### PR TITLE
DRAFT PR: LRA overview doc

### PR DESCRIPTION
**Pull request by ljamen**
I left some questions in Slack. This is the overview doc. I think a guide may be useful as well.

### CODE REVIEW

File: docs/mp/lra/01_introduction.adoc [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+:quickstart-example: {helidon-tag}/examples/quickstarts/helidon-quickstart-mp
+:lra-spec: https://github.com/eclipse/microprofile-lra/blob/master/spec/src/main/asciidoc/microprofile-lra-spec.adoc
+:helidon-mp-lra-example: {helidon-tag}/examples/microprofile/lra
+:common-page-prefix-inc: ../../shared/lra/common_shared.adoc
+:mp-pages-ref-prefix: mp/lra
+:mp-using-lra-ref: {mp-pages-ref-prefix}/02_using-lra.adoc
+:mp-lra-config-ref: {mp-pages-ref-prefix}/03_configuration-with-lra-mp.adoc
+:helidon-variant: MP
+
+Microservices create challenges for data consistency and integrity that necessitate changes in the transaction processing and data patterns used by them. Helidon MP now has its own LRA implementation that adds additional features to the existing MicroProfile LRA specification.
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:34]: Is "LRA" a well-known acronym to readers of this?  (I know that my own brain wants to substitute "Least Recently…" and then I can't complete the "A". 😄 )

Comment (ljnelson) [reply docs/mp/lra/01_introduction.adoc:34]: I see that it's in the title, but depending on where and when the title is rendered and on what platforms and such you may want to use "LRA (Long Running Activities)" instead to start.

File: docs/mp/lra/01_introduction.adoc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+
+== Overview
+As stated in the MicroProfile 4.0 specification, link:{lra-spec}[MicroProfile LRA], "[...] introduces an API for loosely-coupled services to coordinate long running activities in such a way as to guarantee a globally consistent outcome without the need to take locks on data.” 
+
+Helidon MP supports the features described in the Microprofile spec, but also includes additional features:
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:40]: Is "spec" OK to use, given that you previously used "specification"?  Nit, obviously.

File: docs/mp/lra/01_introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+
+== Overview
+As stated in the MicroProfile 4.0 specification, link:{lra-spec}[MicroProfile LRA], "[...] introduces an API for loosely-coupled services to coordinate long running activities in such a way as to guarantee a globally consistent outcome without the need to take locks on data.” 
+
+Helidon MP supports the features described in the Microprofile spec, but also includes additional features: 
+
+
+* Support for messaging with AQ and Kafka
+* DB persistence of tx/LRA records and config for messaging
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:44]: "tx"; we all know but this is documentation 😄

File: docs/mp/lra/01_introduction.adoc [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+Microservices create challenges for data consistency and integrity that necessitate changes in the transaction processing and data patterns used by them. Helidon MP now has its own LRA implementation that adds additional features to the existing MicroProfile LRA specification.
+
+
+== Overview
+As stated in the MicroProfile 4.0 specification, link:{lra-spec}[MicroProfile LRA], "[...] introduces an API for loosely-coupled services to coordinate long running activities in such a way as to guarantee a globally consistent outcome without the need to take locks on data.” 
+
+Helidon MP supports the features described in the Microprofile spec, but also includes additional features: 
+
+
+* Support for messaging with AQ and Kafka
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:43]: Is "AQ" a well-known acronym?

File: docs/mp/lra/01_introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* DB persistence of tx/LRA records and config for messaging
+* Future JSON support (if desired). Currently Helidon supports relational/SQL, but in the future we may provide flatfile or other persistence support.
+* Support for high availability (HA) and Kubernetes 
+
+== Messaging Support with AQ and Kafka
+
+The support for MicroProfile Messaging specification mimics the behavior of JAX-RS path as closely as possible.
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:50]: What is "JAX-RS path"?

File: docs/mp/lra/01_introduction.adoc [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+* Support for messaging with AQ and Kafka
+* DB persistence of tx/LRA records and config for messaging
+* Future JSON support (if desired). Currently Helidon supports relational/SQL, but in the future we may provide flatfile or other persistence support.
+* Support for high availability (HA) and Kubernetes 
+
+== Messaging Support with AQ and Kafka
+
+The support for MicroProfile Messaging specification mimics the behavior of JAX-RS path as closely as possible. 
+For example, any aspect that applies to LRA-related JAX-RS headers applies to messaging properties as well.
```
Comment (ljnelson) [inline docs/mp/lra/01_introduction.adoc:51]: Do we know what a "LRA-related JAX-RS header" might be yet?

**Review by ljnelson — CHANGES_REQUESTED** [review]
I'm going to stop my comments where they are and press the **Request changes** button now just because I think overall some of the assumptions with regard to acronyms and overall audience familiarity may be suspect here.  If this is supposed to be documentation for people who are already familiar with all the technologies, then I guess I'd ask: why is this in `01_introduction.adoc`?  If, on the other hand, people are not expected to know what "AQ" or "LRA" or "JAX-RS path" are, then most of my "requested changes" (how stuffy!) can probably be implemented by briefly explaining the various terms and acronyms involved.  So: hopefully not too much effort.

**Comment by ljnelson** [comment]
And, going out on a limb here, but guessing that @paulparkinson probably is The Person™ here. 😄

File: docs/mp/lra/01_introduction.adoc (hunk 5 above)
Comment (paulparkinson) [reply docs/mp/lra/01_introduction.adoc:58]: @Path

File: docs/mp/lra/01_introduction.adoc (hunk 3 above)
Comment (paulparkinson) [reply docs/mp/lra/01_introduction.adoc]: yes, let's lose the tx :)

File: docs/mp/lra/01_introduction.adoc (hunk 4 above)
Comment (paulparkinson) [reply docs/mp/lra/01_introduction.adoc]: agreed, good idea to spell it out Oracle Advanced Queueing (AQ) JMS

File: docs/mp/lra/01_introduction.adoc (hunk 1 above)
Comment (paulparkinson) [reply docs/mp/lra/01_introduction.adoc]: And to that point it's actually Long Running Actions. That was my mistake (older specs called them Long Running Activities and old habits die hard).

File: docs/mp/lra/01_introduction.adoc (hunk 6 above)
Comment (paulparkinson) [reply docs/mp/lra/01_introduction.adoc:59]: This relates to the "Despite being in the "ws.ra" package, the `org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER` value
 will be reused as the key for AQ message properties, and Kafka K/V pairs will be used for LRA functionality."  part above.   
 
 We could state that here instead and replace "Despite being in the "ws.ra" package" with "For example,"

**Comment by ljamen** [comment]
See Daniel's PR.

**Pull request closed without merging**